src/plugins/ui_feature/split_to_columns.ts

```diff
@@ -0,0 +1,221 @@
+  static getters = ["getAutomaticSeparator", "isSeparatorInSelection"] as const;
+
+  allowDispatch(cmd: Command) {
+    switch (cmd.type) {
+      case "SPLIT_TEXT_INTO_COLUMNS":
+        return this.chainValidations(
+          () => this.checkValidations(cmd, this.checkSingleColSelected, this.checkNonEmptySelector),
+          this.checkNotOverwritingContent
+        )(cmd);
+    }
+    return CommandResult.Success;
+  }
+
+  handle(cmd: Command) {
+    switch (cmd.type) {
+      case "SPLIT_TEXT_INTO_COLUMNS":
+        this.splitIntoColumns(cmd);
+        break;
+    }
+  }
+
+  getAutomaticSeparator(): string {
+    const cells = this.getters.getSelectedCells();
+    for (const cell of cells) {
+      if (cell.value && typeof cell.value === "string") {
+        const separator = this.getAutoSeparatorForString(cell.value);
+        if (separator) {
+          return separator;
+        }
+      }
+    }
+    return " ";
+  }
+
+  private getAutoSeparatorForString(str: string): string | undefined {
+    const separators = [NEWLINE, ";", ",", " ", "."];
+    for (const separator of separators) {
+      if (str.includes(separator)) {
+        return separator;
+      }
+    }
+    return;
+  }
+
+  isSeparatorInSelection(separator: string) {
+    const cells = this.getters.getSelectedCells();
+    for (const cell of cells) {
+      if (cell.formattedValue.includes(separator)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private splitIntoColumns({ separator, addNewColumns }: SplitTextIntoColumnsCommand) {
+    const fullSelection = this.getters.getSelectedZones();
+    const sheetId = this.getters.getActiveSheetId();
+    // Should never happen because of the allowDispatch, but just making sure
+    if (fullSelection.length === 0 || fullSelection[0].left !== fullSelection[0].right) {
+      throw new Error("More than one column selected");
+    }
+    const selection = fullSelection[0];
+    const col = selection.left;
+    const splitted = this.getSplittedCols(selection, separator);
+    if (addNewColumns) {
+      this.addColsToAvoidCollisions(selection, splitted);
+    }
+    this.removeMergesInSplitZone(selection, splitted);
+    this.addColumnsToNotOverflowSheet(selection, splitted);
+
+    for (let i = 0; i < splitted.length; i++) {
+      const row = selection.top + i;
+      const splittedContent = splitted[i];
+      if (splittedContent.length === 1) {
+        continue;
+      }
+      const mainCell = this.getters.getCell({ sheetId, col, row });
+      for (const [index, content] of splittedContent.entries()) {
+        this.dispatch("UPDATE_CELL", {
+          sheetId,
+          col: col + index,
+          row,
+          content,
+          format: undefined,
+          style: mainCell?.style || null,
+        });
+      }
+    }
+  }
+
+  private getSplittedCols(zone: Zone, separator: string): string[][] {
+    if (!separator) throw new Error("Separator cannot be empty");
```

Comment:
I think it's definitely worth being on multi line in this case, I did not realize it was throwing an error